src/v/cloud_storage/partition_manifest.cc

```diff
@@ -2033,15 +2033,18 @@ ss::future<iobuf> partition_manifest::serialize_buf() const {
     return ss::make_ready_future<iobuf>(to_iobuf());
 }
 
-void partition_manifest::serialize_json(std::ostream& out) const {
+void partition_manifest::serialize_json(
+  std::ostream& out, bool include_segments) const {
     serialization_cursor_ptr c = make_cursor(out);
     serialize_begin(c);
-    while (!c->segments_done) {
-        serialize_segments(c);
-    }
-    serialize_replaced(c);
-    while (!c->spillover_done) {
-        serialize_spillover(c);
+    if (include_segments) {
```

Comment:
I think that for recovery you only need to log last segment in the manifest. The `partition_manifest` has `last_segment` method.  It also makes sense to log start offset and last offset of the manifest.

Reply:
@Lazin I've added this, could you have a look please?